format.c

```diff
@@ -4356,6 +4477,39 @@ format_window_name(struct format_expand_state *es, const char *fmt)
 	return (xstrdup("0"));
 }
 
+static int
+format_cmp_window(const void *a0, const void *b0)
+{
+	const struct winlink *const	*a = a0;
+	const struct winlink *const	*b = b0;
+	const struct window		*wa = (*a)->window;
+	const struct window		*wb = (*b)->window;
+	int				 result = 0;
+
+	switch (sc.field) {
+	case FORMAT_LOOP_BY_INDEX:
+		result = wa->id - wb->id;
```

Comment:
Due to this, windows are sorted by id by default, not by index, on the status bar

Reply:
Fixing this in https://github.com/tmux/tmux/pull/4551